dev/io.openliberty.netty.internal.tls.impl/src/io/openliberty/netty/internal/tls/impl/NettyTlsProviderImpl.java

```diff
@@ -0,0 +1,326 @@
+        connectionInfo.put(Constants.CONNECTION_INFO_DIRECTION, direction);
+        connectionInfo.put(Constants.CONNECTION_INFO_REMOTE_HOST, host);
+        connectionInfo.put(Constants.CONNECTION_INFO_REMOTE_PORT, port);
+        connectionInfo.put(Constants.CONNECTION_INFO_ENDPOINT_NAME, endPoint);
+
+        // PK40641 - handle channel props that do not need JSSEHelper
+        Properties props = null;
+        boolean useJSSEHelper = (null != alias);
+        if (!useJSSEHelper) {
+            // 436920 - check for ssl properties put programmatically on the thread.
+            // Calling into JSSEHelper.getProperties() does this same action
+            try {
+                props = AccessController.doPrivileged(new PrivilegedExceptionAction<Properties>() {
+                    @Override
+                    public Properties run() throws Exception {
+                        return com.ibm.websphere.ssl.JSSEHelper.getInstance().getSSLPropertiesOnThread();
+                    }
+                });
+            } catch (Exception e) {
+                // no ffdc required
+                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                    Tr.debug(tc, "Caught exception looking for on-thread props; e=" + e);
+                }
+            }
+            if (null == props || 0 == props.size()) {
+                // if alias is null, then if keystore and truststore are found then
+                // simply use the existing config properties. If either one is
+                // missing, call into the jssehelper for the default repertoire (null alias)
+                props = null;
+                useJSSEHelper = !properties.containsKey(Constants.SSLPROP_KEY_STORE)
+                        || !properties.containsKey(Constants.SSLPROP_TRUST_STORE);
+            } else {
+                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                    Tr.debug(tc, "Found on-thread ssl properties");
+                }
+            }
+        }
+        if (useJSSEHelper) {
+            try {
+                // Extract the official SSL props based on config information.
+                final String aliasFinal = alias;
+                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                    Tr.debug(tc, "Querying security service for alias=[" + aliasFinal + "]");
+                }
+                props = AccessController.doPrivileged(new PrivilegedExceptionAction<Properties>() {
+
+                    @Override
+                    public Properties run() throws Exception {
+                        return com.ibm.websphere.ssl.JSSEHelper.getInstance().getProperties(aliasFinal, connectionInfo,
+                                null);
+                    }
+                });
+            } catch (
+
+            Exception e) {
+                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
```

Comment:
A general comment on having isDebugEnabled() and tr.debug when catching unexpected exceptions that will stop the server working as customer expects. Errors/Warnings, customer turns trace on?

Reply:
Similar to the next comment, this is based off the SSL creation code in com.ibm.ws.channel.ssl [here](https://github.com/OpenLiberty/open-liberty/blob/e09e937bd8e87ee850c82fc5b6f8a50a736edb28/dev/com.ibm.ws.channel.ssl/src/com/ibm/ws/channel/ssl/internal/SSLChannel.java#L483) used also by channelfw. I created https://github.com/OpenLiberty/open-liberty/issues/24838 to follow up on this a bit more